sympy/printing/pycode.py

```diff
@@ -610,6 +612,109 @@ def _print_CodegenArrayPermuteDims(self, expr):
     def _print_CodegenArrayElementwiseAdd(self, expr):
         return self._expand_fold_binary_op('numpy.add', expr.args)
 
+    def _print_Indexed(self, expr):
+        elem = []
+        for i in range(expr.rank):
+            e = expr.indices[i]
+            if isinstance(e, Add) and e.has(Idx):
+                lower = 0
+                upper = 0
+                for a in e.args:
+                    if isinstance(a, Idx):
+                        lower += a.lower
+                        upper += a.upper
+                    else:
+                        lower += a
+                        upper += a
+                elem.append("%s:%s"%(lower, upper))
+            else:
+                elem.append(self._print(expr.indices[i]))
+        return "%s[%s]" % (self._print(expr.base.label), ', '.join(elem))
+
+    def _print_Idx(self, expr):
```

Comment:
I think `PythonCodePrinter` should behave the same way as `CCodePrinter` whenever possible.